projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-social-media-connections/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+
+/**
+ * Hooks to deal with the social media connections.
+ *
+ * @returns {Function} Social media connection handler.
+ */
+export default function useSocialMediaConnections() {
+	const { editPost } = useDispatch( editorStore );
+
+	const connections = useSelect(
+		select => select( editorStore ).getEditedPostAttribute( 'jetpack_publicize_connections' ),
+		[]
+	);
+
+	return {
+		connections,
+		toggleById: function ( id ) {
+			const filteredConnections = connections.map( connection => ( {
+				...connection,
+				enabled: connection.id === id ? ! connection.enabled : connection.enabled,
+			} ) );
+
+			editPost( {
+				jetpack_publicize_connections: filteredConnections,
+			} );
```

Comment:
```suggestion
			const connectionIndex = connections.findIndex( connection => connection.id === id );
			if ( false !== connectionIndex ) {
				connections[ connectionIndex ].enabled = ! connections[ connectionIndex ].enabled;
			}

			editPost( {
				jetpack_publicize_connections: connections,
			} );
```

As there should only be one connection with the given ID, should we use `findIndex` here?

Reply:
I did a little bit of testing as to why this doesn't work and I think because the object reference for `jetpack_publicize_connections` doesn't change, then the edit doesn't do anything. 

Looking at this again, it doesn't make sense. I was trying to prevent always mapping the whole array, but the optimisation isn't worth the trouble caused by changing a property of an object in an array that is defined as `const`